routes/shopItem.js

```diff
@@ -0,0 +1,24 @@
+const express = require("express");
+const fs = require("fs");
+const resolve = require("path").resolve;
+let id = 0;
+
+shopItemRouter = express.Router();
+
+shopItemRouter.use("/", (req,res,next) => {
+id++;
+req.body.id = id;
+next();
+})
+
+shopItemRouter.post("/", (req, res, next) => {
+    res.json(`${req.method} success`);
+    console.log(req.body);
+    fs.writeFileSync(resolve("./database.json"), JSON.stringify(req.body) + "\n", { flag: "a+" }, error => { console.log(error) })
```

Comment:
Only return a "successful" response when all steps are successfully completed. In this case this means that the body is processed, and the file is written out without an error.

Also `writeFileSync` does not have an error callback. Only the asynchronous version have callback. This function will throw an Error on an error that can be caught with a try-catch. In case of an error the servers should return an "internal error" error code in order to tell the client that the operation might have failed.

Something like this:

```js
try {
  fs.writeFileSync(resolve("./database.json"), JSON.stringify(req.body) + "\n");
  res.json(`${req.method} success`);
} catch (err) {
  console.log(err);
  // TODO: please return a status code 500 here
}
```